### PR TITLE
Remove NaCl code

### LINKED ISSUE #187: rip out nacl code
nacl is obsolete and is just sitting there. @Shpoike has said to get rid of it.

### PR DESCRIPTION
**Pull request by fhomolka**
Resolves #187 
I've briefly tested emscripten, which still seems to work fine (at least with Quake 1).
There is one remnant left over, with is the utf-8 encoding option for packets. Seemingly used only by NaCl, but I wanted to make sure before removing it.

Since I am pretty green to the codebase, I'd appreciate a thorough look into this.
I've made several commits, but they can be squash-merged without issue.

### CODE REVIEW

**Comment by eukara** [comment]
Looks OK to me @Shpoike

File: engine/gl/gl_vidcommon.c [hunk 1]
```diff
@@ -282,11 +282,6 @@ FTEPFNGLACTIVESTENCILFACEEXTPROC qglActiveStencilFaceEXT;
 
 
 #define GLchar char
-#ifdef NACL
-#undef DEBUG
-#elif defined(_DEBUG) && !defined(DEBUG)
-#define DEBUG
```
Comment (Shpoike) [inline engine/gl/gl_vidcommon.c:288]: this looks unintended

Comment (fhomolka) [reply engine/gl/gl_vidcommon.c:288]: You're right, this is wrong. Let me fix this

Comment (eukara) [reply engine/gl/gl_vidcommon.c:288]: Thank you, IRC Spoike said it's OK to merge!

**Pull request merged**